torch/distributed/algorithms/join.py

```diff
@@ -39,9 +39,33 @@ def post_hook(self, is_last_joiner: bool):
         """
         ...
 
+
+class _Joinable(ABC):
+    r"""
+    This defines an abstract base class for joinable classes. A joinable class
+    (inheriting from :class:`_Joinable`) should implement a private
+    ``_join_hook()`` method that returns a :class:`_JoinHook` instance.
+    """
+    @abstractmethod
+    def __init__(self):
+        self._join_config = _JoinConfig.construct_disabled_join_config()
+
+    @abstractmethod
+    def _join_hook(self, **kwargs) -> _JoinHook:
+        r"""
+        Returns a :class:`_JoinHook` instance for the given :class:`_Joinable`.
+
+        Arguments:
+            kwargs (dict): a :class:`dict` containing any keyword arguments
+            to modify the behavior of the join hook at run time; all
```

Comment:
missing indent?

Reply:
Fixed locally; will be there in the next push.